src/Analysis/GWASUIMockup/GWASUI.jsx

```diff
@@ -0,0 +1,483 @@
+  const [form] = Form.useForm();
+  const [current, setCurrent] = useState(0);
+  const [selectedCohort, setSelectedCohort] = useState(undefined);
+  const [selectedConcepts, setSelectedConcepts] = useState([]);
+  const [numOfPC, setNumOfPC] = useState(3);
+  const [selectedPhenotype, setSelectedPhenotype] = useState(undefined);
+  const [selectedCovariates, setSelectedCovariates] = useState([]);
+  const [jobName, setJobName] = useState(undefined);
+  const [showJobSubmissionResult, setShowJobSubmissionResult] = useState(false);
+  const [jobSubmittedRunID, setJobSubmittedRunID] = useState(undefined);
+  const [marinerJobStatus, setMarinerJobStatus] = useState([
+    {
+      runID: 'run-10000',
+      status: 'completed',
+      output: [
+        {
+          name: 'test_null_model_report.html',
+          url: 'https://qa-mickey.planx-pla.net/dashboard/Secure/hapmap-genesis-public-outputs/test_null_model_report.html',
+        },
+        {
+          name: 'test_null_model_invnorm_report.html',
+          url: 'https://qa-mickey.planx-pla.net/dashboard/Secure/hapmap-genesis-public-outputs/test_null_model_invnorm_report.html',
+        },
+        {
+          name: 'test_single_assoc_qq.png',
+          url: 'https://qa-mickey.planx-pla.net/dashboard/Secure/hapmap-genesis-public-outputs/test_single_assoc_qq.png',
+        },
+        {
+          name: 'test_single_assoc_manhattan.png',
+          url: 'https://qa-mickey.planx-pla.net/dashboard/Secure/hapmap-genesis-public-outputs/test_single_assoc_manhattan.png',
+        },
+      ],
+    },
+  ]);
+  const [sources, setSources] = useState([]);
+  const [cohortData, setCohortData] = useState([]);
+
+
+  const onStep3FormSubmit = useCallback((values) => {
+    setNumOfPC(values.numOfPC);
+  }, []);
+
+  const handleDelete = (key) => {
+    const newlySelectedPhenotype = (selectedPhenotype.concept_id === key) ? undefined : selectedPhenotype;
+    const newlySelectedCovariates = selectedCovariates.filter((item) => item.concept_id !== key);
+    setSelectedConcepts(selectedConcepts.filter((item) => item.concept_id !== key));
+    setSelectedPhenotype(newlySelectedPhenotype);
+    setSelectedCovariates(newlySelectedCovariates);
+    form.setFieldsValue({
+      covariates: newlySelectedCovariates.map((val) => val.name),
+      // outcome: newlySelectedPhenotype.name,
+    });
+  };
+
+  const getMissingRatio = (cohortArr) => {
+    const cohortKeys = Object.keys(cohortArr[0]);
+    cohortKeys.shift();
+    const missingRatioArr = [];
+    cohortKeys.forEach(key => {
+      let keyRow = {};
+      cohortArr.forEach(cohort => {
+        if (cohort[key] !== "") {
+          keyRow[key] = (keyRow[key] + 1) || 1;
+        }
+      })
+      missingRatioArr.push(keyRow);
+    });
+    const len = cohortArr.length;
+    const stepArrayData = cohortKeys.map((element, i) => ({
+      concept_id: ((i + 1) * 1000) + i, name: element, n_missing_ratio: 1 - (missingRatioArr[i][element] / len)
```

Comment:
is `missingRatioArr[i][element]` guaranteed to have a value here?

Reply:
function is now only called if parameter is an array with greater than 0 length